content/blog/2023/2023-09-16-create-openfeign-http-client.md

```diff
@@ -0,0 +1,503 @@
+
+### Step 3: Define a client for API endpoint
+We use Feign library `builder` to prepare the client. 
+    
+```java
+final CalculatorService target = Feign.builder().decoder(new JacksonDecoder())
+                                      .target(CalculatorService.class, HOST);
+
+```    
+There are many ways to prepare the client depending on our needs. The code snippet given above is just one of the simple ways to prepare the client. We have registered the decoder used to decode the JSON responses. The decoder can be changed to match the content type of the response returned by the service. We will learn more about [decoders](#integrating-encoderdecoder).
+
+### Step 4: Call the service
+Now let us call `add` method of the the service.
+```java
+final Long result = target.add(firstNumber, secondNumber);
+```
+We notice that calling service with Feign HTTP client is fairly simple compared to other [HTTP clients](https://reflectoring.io/comparison-of-java-http-clients/).
+
+You can see it in action by running `givenTwoNumbersReturnAddition` test in the example code shared on [Github](#example-code).
+
+## Getting Familiar with Annotations
+OpenFeign uses a different set of annotations for defining HTTP requests and their parameters. Here's a table of commonly used OpenFeign annotations with examples:
+
+| Annotation | Description | Example |
+| --- | --- | --- |
+| `@RequestLine` | Specifies the HTTP method and path. | `@RequestLine("GET /resource/{id}")` |
+|`@Headers`|Specifies HTTP headers for the request.|`@Headers("Authorization: Bearer {token}")`|
+|`@QueryMap`|Maps a Map of query parameters to the request.|`@QueryMap Map<String, Object> queryParams`|
+|`@Body`|Sends a specific object as the request body.|`@Body RequestObject requestObject`|
+|`@Param`|Adds a query parameter to the request.|`@Param("id") long resourceId`|
+|`@Path`|Replaces a template variable in the path.|`@Path("id") long resourceId`|
+|`@RequestHeader`|Adds a header to the request.|`@RequestHeader("Authorization") String authToken`|
+|`@Headers`|Specifies additional headers for the request.|`@Headers("Accept: application/json")`|
+
+**These annotations allow us to define and customize OpenFeign client interface, making it easy to interact with remote services using OpenFeign. us can mix and match these annotations based on our specific API requirements.**
+
+## Handling Responses
+Feign also provides a declarative approach to API integration. Instead of manually writing boilerplate code for handling request/response serialization/deserialization or error handling, Feign allows us to define these aspects declaratively using annotations. This not only reduces the amount of code we need to write but also improves readability and maintainability.
```

Comment:
Can you add a code snippet with an example to this section? Without that, it's of little help to the reader.